frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -605,13 +621,52 @@ private ContainerLaunchContext setupContainerLaunchContext(TaskStatus taskStatus
     return launchContext;
   }
 
+  // This function is to convert task's containerPorts from List<Range> format to string format
+  // The string format is "httpPort:80,81,82;sshPort:1021,1022,1023;"
+
+  private String setupPortsEnvironment(TaskStatus taskStatus) {
```

Comment:
>setupPortsEnvironment [](start = 17, length = 21)

should add ut for this function

Reply:
Move the main function into ValueRangeUtils and create UTs for it.

---
In reply to: [173375498](https://github.com/Microsoft/pai/pull/318#discussion_r173375498) [](ancestors = 173375498)